contracts/token/StakableToken.sol

```diff
@@ -45,63 +49,80 @@ contract StakableToken is StandardToken {
     return true;
   }
 
+  function nodeStake(uint256 _value) returns (bool success) {
+    // TODO: Permissions
+    if (registry.isNodeRegistered(msg.sender) == true) revert();
+    require(_value > 0);
+    // TODO: Timelock stake
 
-  /*
-  * Core throttling functions
-  */
-
-  function throttle(address _stakerAddress) returns (bool success) {
-
-    // Get current epoch state of staker
-    uint256 currentEpochTransactionCount = epochTransactionCount[_stakerAddress];
-    uint256 currentStakedAmount = stakedAmount[_stakerAddress];
-
-    // Coefficient is how many transactions per throttle epoch are allowed
-    // TODO: dynamic coefficient calculation
-    // Use velocity and supply to determine this coefficient
-    uint256 throttleCoefficient = currentStakedAmount * 2;
-
-    // Luis: Save gas, just save the coefficient once and reset it when the throttle resets
-    if (currentEpochTransactionCount > throttleCoefficient) {
-      //resetThrottleEpoch(_stakerAddress);
-      return false;
-    } else {
-      epochTransactionCount[_stakerAddress] += 1;
-      return true;
-    }
-
+    balances[msg.sender] -= _value;
+    nodeStakedAmount[msg.sender] += _value;
+    Staked(msg.sender, _value);
+    return true;
   }
 
-  // TO-DO: Figure out incentives for resetting the epoch
-  function resetThrottleEpoch(address _stakerAddress) {
-    //Must stake at least 1 PKT
-    require(stakedAmount[_stakerAddress] > 0);
-    // check if current throttle epoch needs to be reset
-    require(block.number >= throttleResetBlock);
-
-    // TODO: DO FUNCTION
+  function releaseNodeStake(uint256 _value) returns (bool success) {
+    require(_value <= nodeStakedAmount);
```

Comment:
Permissions here should be something like:

`require(balances[msg.sender])`

Reply:
Fixed.